web/modules/custom/collection/src/Plugin/Block/CollectionContentBlock.php

```diff
@@ -87,7 +87,9 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function build() {
     if (empty($this->collection)) {
-      throw new \Exception('The "Collection content" block can only be shown on collection pages.');
+      // If this is not a collection page, return an empty array so that the
+      // block is not rendered.
+      return [];
```

Comment:
This seems like a step backwards, why do we need to relax our error checking? If there is no collection on the route this code should not even be running.